framework/include/multiapps/MultiApp.h

```diff
@@ -117,9 +117,11 @@ class MultiApp : public MooseObject,
    * Calls multi-apps executioners' endStep and postStep methods which creates output and advances
    * time (not the time step; see incrementTStep()) among other things. This method is only called
    * for Picard calculations because for loosely coupled calculations the executioners' endStep and
-   * postStep methods are called from solveStep().
+   * postStep methods are called from solveStep(). This may be called with the optional flag \p
+   * recurse_through_multiapp_levels which may be useful if this method is being called for the
+   * *final* time of program execution
    */
-  virtual void finishStep() {}
+  virtual void finishStep(bool /*recurse_through_multiapp_levels*/ = false) {}
```

Comment:
This guy is doing nothing?

Reply:
oh it is the base, nvm.

Reply:
Should we do this always recursively? Of cause I do not know the implications here ;-) just throwing wrenches.

Reply:
No we should not. We also call this from `incrementStepOrReject` where we do not want to recurse through. But when the master solve is totally finished, then we also call this method, and it is then that we need to recurse through, otherwise we do not finish the steps of multiapp levels farther down than the first level.